src/scripts/main.js

```diff
@@ -1,3 +1,300 @@
+};
+
+function startGame() {
+  startButton.classList.remove('start');
+  startButton.style.outline = 'none';
+  startButton.classList.add('restart');
+
+  if (startButton.innerText === 'Start') {
+    startButton.innerText = 'Restart';
+
+    const startMessage = document.querySelector('.message-start');
+
+    startMessage.classList.add('hidden');
+  } else {
+    scoreCounter = 0;
+    scoreInfo.innerText = scoreCounter;
+
+    const cells = getAllCells();
+
+    cells.forEach(cell => {
+      cell.className = 'field-cell';
+      cell.innerText = '';
+    });
+
+    const looseMessage = document.querySelector('.message-lose');
+    const winMessage = document.querySelector('.message-win');
+
+    if (looseMessage) {
+      looseMessage.classList.add('hidden');
+    };
+
+    if (winMessage) {
+      winMessage.classList.add('hidden');
+    }
+  };
+
+  addRandomNumber(getRandomNumber());
+  addRandomNumber(getRandomNumber());
+};
+
+function getAllCells() {
+  return [...document.querySelectorAll('.field-cell')];
+};
+
+function getFreeCells() {
+  return [...document.querySelectorAll('.field-cell')]
+    .filter(cell => cell.innerText === '');
+};
+
+function moveCells(firsCell, secondCell) {
+  const firstValue = Number(firsCell.innerText);
+  const secondValue = Number(secondCell.innerText);
+
+  if (firstValue !== 0 && secondValue === 0) {
+    firsCell.className = 'field-cell';
+    firsCell.innerText = '';
+    secondCell.innerText = firstValue;
+    secondCell.classList.add(`field-cell--${firstValue}`);
+    actionCounter++;
+  };
+
+  if (
+    firstValue !== 0 && firstValue === secondValue
+    && !blockedCells.includes(firsCell)
+  ) {
+    firsCell.className = 'field-cell';
+    firsCell.innerText = '';
+    secondCell.innerText = firstValue * 2;
+    secondCell.classList.add(`field-cell--${firstValue * 2}`);
+    blockedCells.push(secondCell);
+
+    scoreCounter += Number(secondCell.innerText);
+    scoreInfo.innerText = scoreCounter;
+    actionCounter++;
+
+    if (secondCell.innerText === '2048') {
+      winGame();
+    };
+  };
+};
+
+const moveRight = () => {
+  [...gameField.rows].forEach(row => {
+    const cells = [...row.cells];
+
+    for (let i = 2; i >= 0; i--) {
+      const firsCell = cells[i];
+      const secondCell = cells[i + 1];
+
+      moveCells(firsCell, secondCell);
+    };
+  });
+};
+
+const moveLeft = () => {
+  [...gameField.rows].forEach(row => {
+    const cells = [...row.cells];
+
+    for (let i = 1; i < 4; i++) {
+      const firsCell = cells[i];
+      const secondCell = cells[i - 1];
+
+      moveCells(firsCell, secondCell);
+    };
+  });
+};
+
+const moveUp = () => {
+  const columns = [];
+  const fieldRows = [...gameField.rows];
+
+  for (let i = 0; i < fieldRows.length; i++) {
+    const column = [];
+
+    fieldRows.forEach(row => {
+      column.push([...row.cells][i]);
+    });
+
+    columns.push(column);
+  };
+
+  columns.forEach(column => {
+    for (let i = 1; i < 4; i++) {
```

Comment:
it's better to create constants to avoid `magic` numbers

Reply:
Explain in more detail, please...

Reply:
If you have any questions, please ask them in fe_chat and tag mentors. It will be faster, because your answers in GitHub will be visible to mentor only after you re-request review

"magic numbers": `4` is not entirely clear what the number 4 means, the question arises why not 3?, not 7?, why 4?, where did it come from? It is better to put such numbers in a variable and name them so that it is clear from the name what this number means